src/shim/umq/host_queue.h

```diff
@@ -157,6 +157,19 @@ typedef struct
   //! @note Queue capacity, must be a power of two.
   uint32_t capacity;
 
+  /*
+   * NOTE!!!
+   *  Due to the cache is not cache coherence between host and device.  We have
+   *  to flush the cache of the host queue.
+   *
+   *  Most importantly, the read_index has to be in different cache line
+   *  (64Bytes in linux) than the write_index. Because the read_index might be
+   *  flushed from a different context from kernel driver that is monitoring
+   *  the completed message. While at the same time, the write_index might be 
+   *  being flushed from UMQ.
+   */ 
+  uint64_t padding[6];
+
```

Comment:
@maxzhen @mamin506 would the above comments and code change make sense?

Reply:
Instead of calculating padding manually. I would suggest to use compiler attributes.
For example,
uint32_t write_index __attribute__((aligned(64)));

Reply:
Instead of calculating padding manually. I would suggest to use compiler attributes.
For example,
`uint32_t write_index __attribute((aligned(64)));`

Reply:
let me try this on the device compiler.

Reply:
It might be better to insert padding manually. This data structure is shared b/w firmware and host and compiled separately with different compilers.

Reply:
> It might be better to insert padding manually. This data structure is shared b/w firmware and host and compiled separately with different compilers.

Asked Brian, he also think using padding manually like this looks ok to him.